### DISCUSSION TITLE
Is there a way to output Microsoft Adaptive Card to JointJS paper area?

### DISCUSSION THREAD

**Discussion by taeil00** [clientIO/joint#2177] [Q&A]
### Introduction

I need to output Microsoft Adaptive Card to JointJS paper area.
Can you provide me with a code or any advice for displaying the rendering result of an Adaptive Card, which is composed of <div> tags, in the JointJS Paper area?

* The htmlResult looks like below codes.

```
<div class="ac-container ac-adaptiveCard" ~~~~ />
 <div ~~~~ >
<div ~~~~~>
</div>
```

------------------------------------------------------------------------------------------------------
To use AdaptiveCards we need to install adaptivecards first by using this command. 
npm install adaptivecards 

After that,  this codes make adaptiveCards rendered HTML.
-----------------------------------------------------------------
```js
const AdaptiveCards = require('adaptivecards');
const joint = require('jointjs');

// Adaptive Card JSON data
const adaptiveCardJson = `{
  "type": "AdaptiveCard",
  "body": [
    {
      "type": "TextBlock",
      "text": "Hello",
      "weight": "bolder"
    },
    {
      "type": "TextBlock",
      "text": " Hellow World"
    }
  ]
}`;

const adaptiveCard = new AdaptiveCards.AdaptiveCard();
adaptiveCard.parse(adaptiveCardJson);

const htmlResult = adaptiveCard.render();
```
----------------------------------------------------------------------

How can I show the htmlResult on the jointjs paper area (graph) ??

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
You will need a custom element view (either an HTMLView from the [tasks](https://www.jointjs.com/demos/tasks) demo or use SVG `<foreignObject>`) and render the adaptive card HTML inside the shape (e.g. you can render and append the HTML [here](https://github.com/clientIO/joint/blob/master/demo/html/src/joint.shapes.html.js#L152-L160)). You will have to change the size of the model to match the HTML (this I am not sure where you get the actual size at the moment).

We will create a demo for this, but it may take some time.

**Reply by kumilingus** [reply]
Here's the demo: https://github.com/clientIO/joint/discussions/2214
